### PR TITLE
OCW Next Webhook Updates

### PR DESCRIPTION
**Pull request by abeglova**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots

#### What are the relevant tickets?
https://github.com/mitodl/ocw-studio/issues/1005

#### What's this PR do?
This updates the ocw next webhook to expect data from concourse and not s3.

#### Any background context you want to provide?
This should be tested with
https://github.com/mitodl/ocw-studio/pull/1028

### CODE REVIEW

File: course_catalog/views.py [hunk 1]
```diff
@@ -426,23 +426,21 @@ def handle_exception(self, exc):
...
-                    kwargs={"course_prefixes": [prefix], "force_overwrite": False},
-                )
-        else:
-            log.error("No records found in webhook: %s", request.body.decode())
+        version = content.get("version")
+        prefix = content.get("prefix")
+
+        if prefix is not None and version == "live":
+            get_ocw_next_courses.delay(
+                kwargs={"course_prefixes": [prefix], "force_overwrite": False}
```
[the remark this answers is not in the document]
Comment (mbertrand) [reply course_catalog/views.py:441]: Also, `get_ocw_next_course` is throwing an error:
```
celery_1   | botocore.errorfactory.NoSuchKey: An error occurred (NoSuchKey) when calling the GetObject operation: The specified key does not exist.
```

In my testing I was trying course `11-382-water-diplomacy-spring-2021`, which is the value for the `prefix` key sent in the webhook, but in the S3 bucket the full path is actually `courses/11-382-water-diplomacy-spring-2021/data.json` (and `courses` comes from the site's starter config).  Will make a note on that PR.

**Review by mbertrand — CHANGES_REQUESTED** [review]

**Review by mbertrand — APPROVED** [review]
looks good! 👍   Have `OPEN_DISCUSSIONS_WEBHOOK_KEY` and `OCW_NEXT_WEBHOOK_KEY` been added to salt-ops?

**Comment by abeglova** [comment]
Thanks! I made a salt-ops pr
https://github.com/mitodl/salt-ops/pull/1509

**Review by mbertrand — CHANGES_REQUESTED** [review]
There's still `OCW_NEXT_WEBHOOK_KEY` in settings.py

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**